node/src/components/consensus/protocols/highway/config.rs

```diff
@@ -31,6 +31,10 @@ pub struct Config {
     pub max_execution_delay: u64,
     /// The maximum number of peers we request the same vertex from in parallel.
     pub max_requests_for_vertex: usize,
+    /// The maximum number of dependencies we request per validator in a batch.
+    /// Limits requests per validator in panorama - in order to get a total number of
+    /// requests, multiply by # of validators.
+    pub max_request_batch_size: usize,
```

Comment:
Should we also limit the number of units we are willing to send in a single response?

We also need to make sure responses don't exceed the network message size limit.

Since proposals are much, much larger than other units it might make sense to add the sizes here instead of just the numbers; or an approximation (e.g. total number of hashes).

Reply:
> Should we also limit the number of units we are willing to send in a single response?

I thought about it but the way requests are constructed cannot be distinguished from any other request for dependencies. Meaning, we would need a general mechanism for limiting how much dependency requests a node can handle.

> We also need to make sure responses don't exceed the network message size limit.

The response is regular `Dependency` message, which means we need a general mechanism for that.

> Since proposals are much, much larger than other units it might make sense to add the sizes here instead of just the numbers; or an approximation (e.g. total number of hashes).

That's a good idea but that would be true for any other catch-up mechanism.